contracts/command-batcher/src/encoding/evm.rs

```diff
@@ -0,0 +1,467 @@
+use std::str::FromStr;
+
+use connection_router::msg::Message;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{HexBinary, Uint256};
+use ethabi::{ethereum_types, Token};
+use k256::{elliptic_curve::sec1::ToEncodedPoint, PublicKey};
+use multisig::msg::{Multisig, Signer};
+use sha3::{Digest, Keccak256};
+
+use crate::{
+    error::ContractError,
+    types::{Command, CommandBatch, CommandType, Operator},
+};
+
+impl TryFrom<Message> for Command {
+    type Error = ContractError;
+
+    fn try_from(msg: Message) -> Result<Self, Self::Error> {
+        Ok(Command {
+            ty: CommandType::ApproveContractCall, // TODO: this would change when other command types are supported
+            params: command_params(
+                msg.source_chain,
+                msg.source_address,
+                ethereum_types::Address::from_str(&msg.destination_address).map_err(|e| {
+                    ContractError::InvalidMessage {
+                        reason: format!("destination_address is not a valid EVM address: {}", e),
+                    }
+                })?,
+                msg.payload_hash.as_slice().try_into().map_err(|e| {
+                    ContractError::InvalidMessage {
+                        reason: format!(
+                            "payload_hash length is not a valid keccak256 hash length: {}",
+                            e
+                        ),
+                    }
+                })?,
+            ),
+            id: command_id(msg.id),
+        })
+    }
+}
+
+impl CommandBatch {
+    pub fn new(
+        messages: Vec<Message>,
+        destination_chain_id: Uint256,
+    ) -> Result<Self, ContractError> {
+        let message_ids: Vec<String> = messages.iter().map(|msg| msg.id.clone()).collect();
+
+        let data = Data {
+            destination_chain_id,
+            commands: messages
+                .into_iter()
+                .map(|msg| msg.try_into())
+                .collect::<Result<Vec<Command>, ContractError>>()?,
+        };
+
+        let id = batch_id(&message_ids);
+
+        Ok(Self {
+            id,
+            message_ids,
+            data,
+        })
+    }
+
+    pub fn msg_to_sign(&self) -> HexBinary {
+        let msg = Keccak256::digest(self.data.encode().as_slice());
+
+        // Prefix for standard EVM signed data https://eips.ethereum.org/EIPS/eip-191
+        let unsigned = [
+            "\x19Ethereum Signed Message:\n32".as_bytes(), // Keccek256 hash length = 32
+            msg.as_slice(),
+        ]
+        .concat();
+
+        Keccak256::digest(unsigned).as_slice().into()
+    }
+
+    pub fn encode_execute_data(&self, multisig: Multisig) -> Result<HexBinary, ContractError> {
```

Comment:
```suggestion
    pub fn encode_execute_data(&self, quorum: Uint256, signers: Vec<Signer>) -> Result<HexBinary, ContractError> {
```
cuz the multisig state is another thing that's awkward and irrelavant to this method.